src/matcha_ml/core/core.py

```diff
@@ -186,6 +203,7 @@ def provision(
         MatchaError: If prefix is not valid.
         MatchaError: If region is not valid.
     """
+    zenml_version_is_supported()
```

Comment:
Does this need to be in other places too?

Reply:
I'm not sure. If you use `get` you're just grabbing things from remote and if you use `destroy` I don't suppose you care what version you had as you're just annihilating everything.

What do you think?